### PR TITLE
feat(ci): pr diff

### PR DESCRIPTION
**Pull request by samugi**
### Summary

add an action to enable executing the `/prdiff <other-pr-url>` command, to get the diff between the current PR's changes and the other PR's (effectively this outputs the diff of the PRs' diffs, using: https://github.com/samugi/gh-compr).

This gives a quick overview of the differences between two PRs and is meant to facilitate the process of reviewing cherry-picks.

### Checklist

- [x] (no) The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [x] (no) There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-4999

### CODE REVIEW

File: .github/workflows/pr-diff.yml
Comment (dndx) [inline .github/workflows/pr-diff.yml:56]: Anyone this external dependency can be pinned?

**Review by dndx — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-diff.yml
Comment (ADD-SP) [inline .github/workflows/pr-diff.yml:61]: Is this a potential shell injection? Feel free to dismiss my review if I'm wrong.

**Review by ADD-SP — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-diff.yml
Comment (dndx) [inline .github/workflows/pr-diff.yml:56]:
```suggestion
        gh extension install samugi/gh-compr --pin "3785a2d3270c52164fb1f7f63bd3c5df66bedead"
```

We usually pin by commit hash to ensure integrity of our supply chain.

**Review by dndx — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-diff.yml
Comment (dndx) [inline .github/workflows/pr-diff.yml:38]: Will there be any risk of command injections? Like

```
/prdiff https://github.com/xxx `do something evil`
```

This may not be a practical example, but we should be extra careful when processing untrusted input (such as these from comments). 

@curiositycasualty could you help to double check here as well?

Comment (samugi) [reply .github/workflows/pr-diff.yml:38]: @dndx I just had a similar discussion with @ADD-SP, it should not be a major concern since we limit execution of this workflow to members, owners, maintainers: https://github.com/Kong/kong/pull/13401/files#diff-84cd7992011f4a1f7fe5735c840973ddeecc24d45d25a978b5f1c1709ef7707cR16-R24

Still, we decided it'd be a good idea to validate the input. I just added a step that checks that we are only passing a valid and expected URL https://github.com/Kong/kong/pull/13401/files#diff-84cd7992011f4a1f7fe5735c840973ddeecc24d45d25a978b5f1c1709ef7707cR43-R61

Comment (curiositycasualty) [reply .github/workflows/pr-diff.yml:38]: Leveraging `env:` to lean on bash/shell escaping is what mitigates injection in actions workflows. So this looks as-good-as-can-be to me.

Comment (kikito) [reply .github/workflows/pr-diff.yml:38]: Marking this as resolved given @curiositycasualty 's comment.

**Comment by kikito** [comment]
@outsinre , could you review this? It should help doing CE-> EE cherrypicks

**Comment by outsinre** [comment]
> @outsinre , could you review this? It should help doing CE-> EE cherrypicks

Let me check.

**Review by outsinre — APPROVED** [review]

**Pull request merged**
